### PR TITLE
[CALCITE-3632] Add IntersectToIntersectUnify Rule in SubstitutionVisitor

### PR DESCRIPTION
**Pull request by xy2953396112**
JIRA:https://issues.apache.org/jira/browse/CALCITE-3632

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/MaterializationTest.java [hunk 1]
```diff
@@ -2857,6 +2858,26 @@ private void checkSatisfiable(RexNode e, String s) {
     checkMaterialize(mv, query);
   }
 
+  @Test public void testIntersectToIntersect0() {
+    final String mv = ""
+        + "select \"deptno\" from\n"
+        + "\"emps\" intersect select \"deptno\"  from \"depts\"";
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/MaterializationTest.java]: Redundant spaces ?

File: core/src/test/java/org/apache/calcite/test/MaterializationTest.java [hunk 2]
```diff
@@ -2857,6 +2858,26 @@ private void checkSatisfiable(RexNode e, String s) {
...
+    final String query = ""
+        + "select \"deptno\" from\n"
+        + "\"depts\" intersect select \"deptno\" from \"emps\"";
+    checkMaterialize(mv, query, true);
+  }
+
+  @Test public void testIntersectToIntersect1() {
```
Comment (jinxing64) [inline core/src/test/java/org/apache/calcite/test/MaterializationTest.java]: How about format like below:
```
select deptno from emps
intersect
select deptno  from depts;

**Comment by jinxing64** [comment]
Just minor comments ~
BTW, this is another case that tests added just for SubstitutionVisitor ~
If committer starts review this one, it will be great that https://github.com/apache/calcite/pull/1560 can also be taken into consideration.

File: core/src/test/java/org/apache/calcite/test/MaterializationTest.java (hunk 2 above)
Comment (xy2953396112) [reply core/src/test/java/org/apache/calcite/test/MaterializationTest.java]: thanks, update code.

**Pull request closed without merging**